app/services/ajax.ts

```diff
@@ -286,6 +288,18 @@ export default class IreneAjaxService extends Service {
     const response = await fetch(finalUrl, fetchOptions);
 
     if (!response.ok) {
+      // Handle rate limit (429) responses
+      if (response.status === 429) {
+        const headersObj: Record<string, string> = {};
+        response.headers.forEach((value, key) => {
+          headersObj[key] = value;
+        });
+
+        const requestData = { url: finalUrl };
+
+        this.rateLimit.handleResponse(this, headersObj, requestData);
```

Comment:
Do we need to continue processing the response after handling the rate limit error?